### ISSUE TITLE
Events are being counted once per assay when splitting by result

### ISSUE THREAD

**Issue by spalladino** [instedd/notifiable-diseases#27]
In the trendline, if events are chosen to be split by result, one event per each assay will be displayed.

For instance, if there is a single event, with three assays:

```
condition: "mtb", result: "positive"
condition: "rif", result: "negative"
condition: "inh", result: "negative"
```

A simple trendline will display a single data point. A trendline split by result will display one positive and two negatives. This applies to both tests and encounters.

Only the result for the condition of the current report should be taken into consideration.

**Comment by matiasgarciaisaia** [comment]
I think this is a cdx-api-elasticsearch issue - not NNDD's.

NNDD's asking this query:

```
{
  "page_size": 0,
  "test.assays.condition": "mtb",
  "group_by": [
    "year(test.start_time)",
    "test.assays.result"
  ],
  "result": [
    "positive",
    "negative",
    "n/a"
  ]
}
```

I think it has all the relevant info.

The problem is the query cdx-api-elasticsearch is generating. Here are both versions - the current one on the left, the fixed one on the right: https://www.diffchecker.com/cbxzad0o

cdx-api-elasticsearch should be aware that the `group_by` depends on the filters (ie, `test.assays.results` is a sibling of `test.assays.condition` on an array)  and make the relevant modifications in the query.

**Comment by macoca** [comment]
I see your point. This sanity check should be automatically made for all the nested fields included in the filter.

It might be solved along with instedd/cdx#403

**Comment by pmallol** [comment]
Ok in Version: 0.11-pre9
